sdk/samples/iot/paho_iot_hub_pnp_component_sample.c

```diff
@@ -99,911 +53,859 @@ static az_span start_time_span;
-static void connect_device(void);
-static void subscribe(void);
+// IoT Hub Telemetry Values
+static const az_span working_set_name = AZ_SPAN_LITERAL_FROM_STR("workingSet");
+
+static iot_sample_environment_variables env_vars;
+static az_iot_hub_client hub_client;
+static MQTTClient mqtt_client;
+static char mqtt_client_username_buffer[512];
+static pnp_mqtt_message mqtt_message;
 
 //
-// Messaging functions
+// Functions
 //
-static void mqtt_publish_message(char* topic, az_span payload, int qos);
-static void mqtt_receive_message(void);
-static int on_received(char* topicName, int topicLen, MQTTClient_message* message);
-static void send_device_serial_number(void);
+static void create_and_configure_mqtt_client(void);
+static void connect_mqtt_client_to_iot_hub(void);
+static void subscribe_mqtt_client_to_iot_hub_topics(void);
+static void initialize_components(void);
 static void send_device_info(void);
-static void send_telemetry_messages(void);
-static void send_twin_get_message(void);
-static void handle_twin_message(
-    MQTTClient_message* message,
-    az_iot_hub_client_twin_response* twin_response);
+static void send_device_serial_number(void);
+static void request_device_twin_document(void);
+static void receive_messages(void);
+static void disconnect_mqtt_client_from_iot_hub(void);
+
+static void mqtt_publish_message(const char* topic, az_span payload, int qos);
+static void mqtt_receive_message(void);
+static void on_message_received(char* topic, int topic_len, const MQTTClient_message* message);
+
+// Device Twin functions
+static void handle_device_twin_message(
+    const az_span twin_message_span,
+    const az_iot_hub_client_twin_response* twin_response);
+
+// Command functions
 static void handle_command_message(
-    MQTTClient_message* message,
-    az_iot_hub_client_method_request* command_request);
+    const az_span command_message_span,
+    const az_iot_hub_client_method_request* command_request);
+static az_result temp_controller_process_command(
+    const az_iot_hub_client_method_request* command_request,
+    az_span component_name,
+    az_span command_name,
+    az_span command_payload,
+    pnp_mqtt_message* mqtt_message,
+    az_iot_status* status);
+
+// Telemetry functions
+static void send_telemetry_messages(void);
+static az_result temp_controller_get_telemetry_message(pnp_mqtt_message* message);
+
+// Callbacks
+static az_result append_string(az_json_writer* jw, void* value);
+static az_result append_json_token(az_json_writer* jw, void* value);
+static void property_callback(
+    az_span component_name,
+    const az_json_token* property_name,
+    az_json_reader property_value,
+    int32_t version,
+    void* user_context_callback);
 
 int main(void)
+{
+  create_and_configure_mqtt_client();
+  LOG_SUCCESS("Client created and configured.");
+
+  connect_mqtt_client_to_iot_hub();
+  LOG_SUCCESS("Client connected to IoT Hub.");
+
+  subscribe_mqtt_client_to_iot_hub_topics();
```

Comment:
nit: should this be
```suggestion
  subscribe_mqtt_client_to_iot_pnp_topics();
```
curious on your opinion

Reply:
hmmm.  i'm not sure.  ultimately, it is connecting to the hub.  @ericwol-msft and @jspaith what do you think?

Reply:
not a big deal feel free to ignore for this PR